### PR TITLE
Updating MySQL, moving it commons

### PR DESCRIPTION
**Pull request by komal1204**
Hi @yunho-oh 

This PR moves MySQL to commons. 

Presently, webserver uses ubuntu base image; this will be changed to debian in a new PR. Building webserver workload and common mysql for all architecture will be included in the new PR. 

If this is merged, we can delete  PR #241 .

Once this PR is merged, #261 and #265 can be merged too.

### CODE REVIEW

**Comment by yunho-oh** [comment]
Hi @komal1204 

Thank you. I will look into this PR as a priority and get back to you soon.

**Comment by yunho-oh** [comment]
Hi @komal1204, 

First off, if we want to upload mariadb-10.3 instead of mysql, we should change all the comments that say mysql and the docker tag (cloudsuite/mysql:mariadb-10.3).
I will leave relevant comments in the 'Files changed' category.

File: .travis.yml [hunk 1]
```diff
@@ -61,6 +61,9 @@ env:
   # - DH_REPO="cloudsuite/cassandra" IMG_TAG="latest" DF_PATH="./commons/cassandra/3.11.6"
   - DH_REPO="cloudsuite/cassandra" IMG_TAG="3.11.6" DF_PATH="./commons/cassandra/3.11.6" DBX_PLATFORM=linux/amd64,linux/arm64
 
+  # MySQL
```
Comment (yunho-oh) [inline .travis.yml:64]: Please change it to MariaDB.

File: benchmarks/web-serving/db_server/Dockerfile [hunk 2]
```diff
@@ -1,30 +1,12 @@
-FROM ubuntu:14.04
+FROM cloudsuite/mysql:mariadb-10.3
```
Comment (yunho-oh) [inline benchmarks/web-serving/db_server/Dockerfile:1]: Please change it to cloudsuite/mariadb:10.3

File: benchmarks/web-serving/db_server/Dockerfile [hunk 3]
```diff
@@ -1,30 +1,12 @@
...
 LABEL maintainer="Mark Sutherland <mark.sutherland@epfl.ch>"
 
 ENV root_password root
-ENV web_host web_server 
 
-# Install MySQL
-RUN echo mysql-server mysql-server/root_password password ${root_password} | sudo debconf-set-selections
-RUN echo mysql-server mysql-server/root_password_again password ${root_password} | sudo debconf-set-selections
```
Comment (yunho-oh) [inline benchmarks/web-serving/db_server/Dockerfile:9]: I think the above two lines should be changed following the changes with mariadb as well.

File: benchmarks/web-serving/db_server/Dockerfile [hunk 4]
```diff
@@ -1,30 +1,12 @@
...
-
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-	mysql-server
-
-# Allow it to listen from outer world
-RUN sed -i -e"s/^bind-address\s*=\s*127.0.0.1/bind-address = 0.0.0.0/" /etc/mysql/my.cnf
```
Comment (yunho-oh) [inline benchmarks/web-serving/db_server/Dockerfile:15]: Please check if it is still necessary if we use mariadb.

File: benchmarks/web-serving/db_server/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (yunho-oh) [inline benchmarks/web-serving/db_server/entrypoint.sh:1]: Please double-check all the lines referencing mysql if it works after you change the name of db to mariadb.

File: commons/mysql/files/execute.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (yunho-oh) [inline commons/mysql/files/execute.sh:1]: Please double-check all the lines referencing mysql if it works after you change the name of db to mariadb.

**Comment by komal1204** [comment]
> Hi @komal1204,
> 
> First off, if we want to upload mariadb-10.3 instead of mysql, we should change all the comments that say mysql and the docker tag (cloudsuite/mysql:mariadb-10.3).
> I will leave relevant comments in the 'Files changed' category.

Hello,

Thanks for the feedback; we kept it like that because in commons we still have it in the mysql folder; and professor suggested to keep it like that, so do we still change it?

File: benchmarks/web-serving/db_server/Dockerfile (hunk 3 above)
Comment (komal1204) [reply benchmarks/web-serving/db_server/Dockerfile:9]: I am sorry , i didn't understand what to change here, these lines are already removed.

File: benchmarks/web-serving/db_server/Dockerfile [hunk 7]
```diff
@@ -1,30 +1,12 @@
...
 
 ENV root_password root
-ENV web_host web_server 
 
-# Install MySQL
-RUN echo mysql-server mysql-server/root_password password ${root_password} | sudo debconf-set-selections
-RUN echo mysql-server mysql-server/root_password_again password ${root_password} | sudo debconf-set-selections
-
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-	mysql-server
```
[the remark this answers is not in the document]
Comment (komal1204) [reply benchmarks/web-serving/db_server/Dockerfile:12]: @yunho-oh  - I am sorry , i didn't understand what to change here, these lines are already removed.

**Comment by yunho-oh** [comment]
My thinking is that at least we should leave some relevant information about this (in the files and maybe readme.md file as well). Because a benchmark suite would be used a wide range of users and all of them have different views and thoughts. So, if the following is correct, at least we should explicitly say we exploit the mysql feature in mariadb to mitigate the confusion (although the confusion does not affect actual use of the benchmark).

File: benchmarks/web-serving/db_server/Dockerfile (hunk 4 above)
Comment (komal1204) [reply benchmarks/web-serving/db_server/Dockerfile:15]: @yunho-oh - Similarly, here as well, this is already removed, as it was not needed.

File: benchmarks/web-serving/db_server/Dockerfile (hunk 3 above)
Comment (yunho-oh) [reply benchmarks/web-serving/db_server/Dockerfile:9]: Sorry, never mind.

File: benchmarks/web-serving/db_server/Dockerfile (hunk 7 above)
[the remark this answers is not in the document]
Comment (yunho-oh) [reply benchmarks/web-serving/db_server/Dockerfile:12]: Sorry, never mind. My comment is wrong.

File: benchmarks/web-serving/db_server/Dockerfile (hunk 4 above)
Comment (yunho-oh) [reply benchmarks/web-serving/db_server/Dockerfile:15]: Sorry, never mind. My comment is wrong.

**Comment by komal1204** [comment]
> My thinking is that at least we should leave some relevant information about this (in the files and maybe readme.md file as well). Because a benchmark suite would be used a wide range of users and all of them have different views and thoughts. So, if the following is correct, at least we should explicitly say we exploit the mysql feature in mariadb to mitigate the confusion (although the confusion does not affect actual use of the benchmark).

Yes, this is a good idea, can include this information in the README . Will do that. Apart from this, is the PR good to be merged?

**Comment by yunho-oh** [comment]
I will keep looking at other parts of this PR. I will try it and leave comments ASAP.  
But (still and always) I have other tasks at the same time. 
I kindly ask you for understanding it.

**Comment by komal1204** [comment]
> I will keep looking at other parts of this PR. I will try it and leave comments ASAP.
> But (still and always) I have other tasks at the same time.
> I kindly ask you for understanding it.

Sure thanks, no worries , take your time.

**Comment by yunho-oh** [comment]
Hi @komal1204,

I tried to run the web serving benchmark. I could build the db_server image with this pull request.
Currently, the web serving benchmark does not run. 
I tried to see what happens in the db_server first. When I run the db_server container without running it in background, I saw the following log.

**Stopping MariaDB database server: mysqldERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)**

Could you please check if it is still fine or not?

**Comment by komal1204** [comment]
> Hi @komal1204,
> 
> I tried to run the web serving benchmark. I could build the db_server image with this pull request.
> Currently, the web serving benchmark does not run.
> I tried to see what happens in the db_server first. When I run the db_server container without running it in background, I saw the following log.
> 
> **Stopping MariaDB database server: mysqldERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)**
> 
> Could you please check if it is still fine or not?

Hello, 
Yes, this message is displayed, but if you exec into the dbserver, and run service mysql status or simple mysql commands, it works. I re-ran the workload again and it was working. Another thing to ensure is that there should be nothing running on MySQL port. 
Let me know if there are any issues still in running the workload.
Thanks,
Komal

**Comment by komal1204** [comment]
> > Hi @komal1204,
> > I tried to run the web serving benchmark. I could build the db_server image with this pull request.
> > Currently, the web serving benchmark does not run.
> > I tried to see what happens in the db_server first. When I run the db_server container without running it in background, I saw the following log.
> > **Stopping MariaDB database server: mysqldERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)**
> > Could you please check if it is still fine or not?
> 
> Hello @yunho-oh 
> Yes, this message is displayed, but if you exec into the dbserver, and run service mysql status or simple mysql commands, it works. I re-ran the workload again and it was working.
In the screenshot, the last line says mysql is running. Another thing to ensure is that there should be nothing running on MySQL port.
> Let me know if there are any issues still in running the workload.
> Thanks,
> Komal

image

**Comment by yunho-oh** [comment]
OK. It works for me now (I don't know what I made a mistake last time). Let's merge this PR. Please make sure that README is updated.

**Review by yunho-oh — APPROVED** [review]
Please make sure that README is updated later.

**Comment by komal1204** [comment]
> Please make sure that README is updated later.

thanks, yes will update the README to mention that we are installing Mariadb.

**Comment by yunho-oh** [comment]
If none of your team members is authorized to merge the source code, we will do it asap.

**Comment by yunho-oh** [comment]
Hi @komal1204,

I missed saying one more thing: please make sure that this PR passes the travisCI to merge it asap.

**Comment by komal1204** [comment]
Hello @yunho-oh 

The travis build is failing because dbserver of webserving workload uses the new mysql image which is also a part of this pr.
#4 [1/5] FROM docker.io/cloudsuite/mysql:mariadb-10.3
#4 resolve docker.io/cloudsuite/mysql:mariadb-10.3
#4 resolve docker.io/cloudsuite/mysql:mariadb-10.3 0.2s done
#4 ERROR: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed

Do suggest what should be done to avoid this or if this can be ignored to merge it?

**Comment by komal1204** [comment]
> The travis build is failing because dbserver of webserving workload uses the new mysql image which is also a part of this pr.
> #4 [1/5] FROM docker.io/cloudsuite/mysql:mariadb-10.3
> #4 resolve docker.io/cloudsuite/mysql:mariadb-10.3
> #4 resolve docker.io/cloudsuite/mysql:mariadb-10.3 0.2s done
> #4 ERROR: pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed

the remaining two errors will be fixed as @ajaygk95 has fixed issue #270

**Comment by komal1204** [comment]
Hello @yunho-oh ,

I have removed the webserving changes from this PR and created a new PR #271 for those changes.
Now the travis build should pass without any errors.
Please update the mysql image in dockerhub too so that PR #271 can be merged after this.
Thanks,
Komal

**Review by neo-apz — APPROVED** [review]

**Comment by komal1204** [comment]
Hello, 

Can you please push the updated mysql image too to docker hub; else the travis build will fail for the PR #271 .

Thanks,
Komal

**Pull request merged**